### PR TITLE
Add Metric Store

### PR DESCRIPTION
**Pull request by toddboom**
## Please take a moment to review the questions before submitting the PR

🚫 We only accept PRs to develop branch. If this is an exception, please specify why 🚫

### WHAT is this change about?

This adds a single node Metric Store to the CF Deployment.

### What customer problem is being addressed? Use customer persona to define the problem e.g. Alana is unable to...

As metric support is phased out of Log Cache, users will need a new store for firehose metrics. Metric Store aims to fill this need by connecting directly to the RLP, durably storing all metrics for a foundation, and providing a PromQL-compatible query interface to access them.

### Please provide any contextual information.

https://www.pivotaltracker.com/story/show/167953063

We are just about to cut an updated release (v1.2.0) that should probably be specified in the manifest before this is merged. However, as I anticipate this PR will generate some discussion, we're getting it up here early so that we can start talking about it sooner.

### Has a cf-deployment including this change passed [cf-acceptance-tests](https://github.com/cloudfoundry/cf-acceptance-tests)?

- [ ] YES
- [x] NO

### Does this PR introduce a breaking change? Please take a moment to read through the examples before answering the question.

- [x] YES
- [ ] NO

This change increases VM footprint of cf-deployment.

### How should this change be described in cf-deployment release notes?

Adds a Metric Store instance, which provides operators and users with a durable store of firehose metrics that can be queried via PromQL.

### Does this PR introduce a new BOSH release into the base cf-deployment.yml manifest or any ops-files?

- [x] YES
- [ ] NO

This will add the `metric-store-release` BOSH release.

### Does this PR make a change to an experimental or GA'd feature/component?

- [x] experimental feature/component
- [ ] GA'd feature/component

### Please provide Acceptance Criteria for this change?

An admin user should be able to run the following curl command and see a sucessful response:

```
curl -H "Authorization: $(cf oauth-token)" -sS -G https://my.foundation.pivotal.io/api/v1/query --data-urlencode 'query=cpu`
```

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!

/cc @colins @attack @hdub2 @josephgee @heidmotron

### CODE REVIEW

**Comment by acosta11** [comment]
Unit tests failed due to known issue. Should now be resolved, so closing and re-opening to re-trigger tests.

**Comment by toddboom** [comment]
@acosta11 thanks for retriggering - looks like it still failed. we'll investigate on this end.

**Comment by weymanf** [comment]
Hey @toddboom Looks like unit tests are still failing for this PR. 
The changes you need to make are in this file: 
https://github.com/cloudfoundry/cf-deployment/blob/develop/operations/rename-network-and-deployment.yml

You have to add the instance to this list to change the network name. And we should be good to go.

File: operations/experimental/add-metric-store.yml [hunk 1]
```diff
@@ -1,169 +0,0 @@
-- type: replace
```
Comment (weymanf) [inline operations/experimental/add-metric-store.yml]: Looks like this operation of excluding the prom_scrapper from the metrics-store instance group was not replaced inside of the base manifest. Is this the correct behavior? Does the metrics-store now need to have the prom_scraper addon.

File: operations/experimental/add-metric-store.yml [hunk 2]
```diff
@@ -1,169 +0,0 @@
...
-  path: /addons/name=prom_scraper/exclude?/instance_groups/-
-  value: metric-store
```
Comment (weymanf) [inline operations/experimental/add-metric-store.yml]: Looks like this operation of excluding the prom_scrapper from the metrics-store instance group was not replaced inside of the base manifest. Is this the correct behavior? Does the metrics-store now need to have the prom_scraper addon?

**Comment by julian-zucker** [comment]
Moved to https://github.com/cloudfoundry/cf-deployment/pull/843

**Pull request closed without merging**